receive_stream.go

```diff
@@ -202,7 +222,8 @@ func (s *receiveStream) dequeueNextFrame() {
 
 func (s *receiveStream) CancelRead(errorCode StreamErrorCode) {
 	s.mutex.Lock()
-	completed := s.cancelReadImpl(errorCode)
+	s.cancelReadImpl(errorCode)
+	completed := s.isNewlyCompleted()
```

Comment:
Does this work when CancelRead happens *JUST* after receiving a reset from remote?